packages/editor/src/extensions/code-block/code-block.ts

```diff
@@ -234,11 +234,34 @@ export const CodeBlock = Node.create<CodeBlockOptions>({
         },
       toggleCodeBlock:
         (attributes) =>
-        ({ commands }) => {
-          return commands.toggleNode(this.name, "paragraph", {
-            ...attributes,
-            id: createCodeblockId()
+        ({ commands, state }) => {
+          const { from, to } = state.selection;
+
+          let hasParagraph = false;
+
+          state.doc.nodesBetween(from, to, (node) => {
+            if (node.type.name === "paragraph") {
+              hasParagraph = true;
+            }
           });
```

Comment:
Instead of checking whether selection has paragraph, we can just check whether a codeblock node is active using `isNodeActive` function.

Reply:
Are you saying that the selected text should only be converted if a codeblock node is not active?

I wonder if the behavior should be to convert all selected text to a codeblock, if any one of the selected nodes is of a type other than codeblock?

Reply:
> Are you saying that the selected text should only be converted if there are no active codeblock nodes?

Yes. Because if the cursor is inside a codeblock we'll just convert it into a paragraph. Otherwise, everything gets converted into a codeblock.